### PR TITLE
[ Dy2Static ] convert_call support staticmethod for class.

### PR DESCRIPTION
**Pull request by 2742195759**
### PR types

Others
### PR changes

Others
### Describe

convert_call support staticmethod for class.

### CODE REVIEW

File: python/paddle/fluid/dygraph/dygraph_to_static/convert_operators.py [hunk 1]
```diff
@@ -87,7 +87,10 @@ def convert_while_loop(cond,
     Args:
         cond(Callable): A callable object that returns a boolean variable to control whether to execute the loop body. It takes ``loop_vars`` as arguments.
         body(Callable): A callable object that returns a tuple or list of variables with the same arguments ``loops_vars`` as ``cond`` .
-        loop_vars(list|tuple): A list or tuple of variables passed to ``cond`` and ``body`` .
+        get_args(callable): Get all arguments that needed in true_fn and false_fn.
+        set_args(callable): Update arguments that modified in trure_fn and false_fn.
+        return_name_ids(list[string]): the returned names.
```
Comment (Ligoml) [inline python/paddle/fluid/dygraph/dygraph_to_static/convert_operators.py]: 有默认值的参数是可选参数，应该备注 optional
`return_name_ids(list[string])` --> `return_name_ids(list[string], optional)`

File: python/paddle/fluid/layers/math_op_patch.py [hunk 2]
```diff
@@ -227,12 +227,31 @@ def append(self, var):
...
+        if len(self.shape) > 1:
+            raise TypeError(
+                "Required input var should be 1-D Variable, but received {}".
+                format(self.shape))
+        return self
+
     @static_only
     def pop(self, *args):
         """
          **Notes**:
```
Comment (Ligoml) [inline python/paddle/fluid/layers/math_op_patch.py]: note 不用加粗，格式参考 [注解](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_docs_guidelines_cn.html#zhujie)

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**